Dockerfile

```diff
@@ -1,57 +1,53 @@
-# syntax=docker/dockerfile:1
+# Create pipenv image to convert Pipfile to requirements.txt
+FROM python:3.9-slim as pipenv
 
-# Set base image
+# Copy Pipfile and Pipfile.lock
+COPY Pipfile Pipfile.lock ./
+
+# Install pipenv and convert to requirements.txt
+RUN pip3 install --no-cache-dir --upgrade pipenv; \
+    pipenv requirements > requirements.txt
+
+FROM python:3.9-slim as python-reqs
+
+# Copy requirements.txt from pipenv stage
+COPY --from=pipenv /requirements.txt requirements.txt
+
+# Install gcc for building python dependencies; install TCM dependencies
+RUN apt-get update; \
+    apt-get install -y gcc; \
+    pip3 install --no-cache-dir -r requirements.txt
+
```

Comment:
Only when building the requirements do we require gcc, this installs gcc and installs requirements in a separate image to be discarded, and we copy out the site-packages